rust/processor/src/models/stake_models/current_delegated_voter.rs

```diff
@@ -0,0 +1,370 @@
+                                            delegation_pool_address_from_query_result.clone(),
+                                            delegator_address.clone(),
+                                        ),
+                                        (
+                                            Some(voter.to_string()),
+                                            Some(pending_voter.to_string()),
+                                            table_handle.clone(),
+                                        ),
+                                    );
+                                },
+                                Err(_) => {
+                                    tracing::error!(
+                                        transaction_version = txn_version,
+                                        lookup_key = &table_handle,
+                                        "Failed to get delegation pool address from vote delegation write table handle. You probably should backfill db.",
+                                    );
+                                },
+                            }
+                        }
+                    }
+                }
+            }
+
+            // this write resource indexing is to get active pool to DelegatorPoolBalanceMetadata mapping
+            // as a helper for us to get delegation pool address and delegator address from DelegationPool
+            // with the assumption that when delegator can delegate votes, they must have already staked
+            for wsc in &transaction.info.as_ref().unwrap().changes {
+                if let WriteSetChange::WriteResource(write_resource) = wsc.change.as_ref().unwrap()
+                {
+                    if let Some(map) =
+                        Self::get_active_pool_to_staking_pool_mapping(write_resource, txn_version)?
+                    {
+                        active_pool_to_staking_pool.extend(map);
+                    }
+                }
+            }
+
+            // this write table item indexing is to get delegation pool address and delegator address PK
+            // to cover the edge case where partial voting didn't exist in the pre vote delegation era
+            // where we try to get the PK and check against the DB to check for its existence
+            // if already there, don't overwrite, if not, try to get vote delegation info from mapping pk_to_vote_delegation
+            // if not found, return vote delegation fields as none
+            for wsc in &transaction.info.as_ref().unwrap().changes {
+                if let WriteSetChange::WriteTableItem(write_table_item) =
+                    wsc.change.as_ref().unwrap()
+                {
+                    if let Some(pk) = Self::get_pk_from_write_table_item(
+                        write_table_item,
+                        &active_pool_to_staking_pool,
+                    )? {
+                        if let Some((maybe_voter, maybe_pending_voter, table_handle)) =
```

Comment:
Actually should this be flipped? If the pk _doesn't_ exist in the post-vote delegation era, then check if there is a row in DB that already exists. If it doesn't exist, then create one. and the voter/pending_voter should be the delegator address.